src/main/java/org/primefaces/extensions/renderer/CommandButtonSingleClickRenderer.java

```diff
@@ -57,13 +54,17 @@ protected String getToggleJS(final String widgetVar, final boolean enabled) {
         return String.format("var w=PF('%s');if(w){w.%sable();};", widgetVar, enabled ? "en" : "dis");
     }
 
-    protected String originalAttributeValue(final CommandButton button, final String current, final String key) {
-        String value = (String) button.getAttributes().get(key);
-        if (value != null) {
-            return value;
+    protected String getAttributeValue(final FacesContext context, final CommandButton button, final String attribute) {
+        String key = "CommandButtonSingleClickRenderer:" + attribute;
+        ValueExpression ve = (ValueExpression) button.getAttributes().get(key);
+        if (ve == null) {
+            ve = button.getValueExpression(attribute);
+        }
+        if (ve != null) {
+            button.getAttributes().put(key, ve);
+            return (String) ve.getValue(context.getELContext());
```

Comment:
I think we must throw in `buildDomEvent` as used in the `commandButton` renderer.

Reply:
Agreed.

Reply:
Sorry for the fuzz, it was too late to code yesterday 😄. `buildDomEvent` will be handled when we call `super.encodeMarkup(context, button);`.